rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2184,6 +2184,9 @@ check_and_attach_condition(
     }
   }
 
+  // No conditions are available. Set the triggered flag of the wait_set to false.
```

Comment:
Should we append a note here that we've already locked `wait_set_data->condition_mutex` prior to calling `check_and_attach_condition`?

Reply:
Done